packages/data-provider/src/controllers/submittedDataController.ts

```diff
@@ -0,0 +1,120 @@
+import { NextFunction, Request, Response } from 'express';
+import * as _ from 'lodash-es';
+
+import { BaseDependencies } from '../config/config.js';
+import submittedDataService from '../services/submittedDataService.js';
+import { BadRequest, NotFound } from '../utils/errors.js';
+import { isEmptyString } from '../utils/formatUtils.js';
+import { SubmittedDataPaginatedResponse } from '../utils/types.js';
+
+const controller = (dependencies: BaseDependencies) => {
+	const service = submittedDataService(dependencies);
+	const { logger } = dependencies;
+	const LOG_MODULE = 'SUBMITTED_DATA_CONTROLLER';
+	const defaultPage = 1;
+	const defaultPageSize = 20;
+	return {
+		getSubmittedDataByCategory: async (
+			req: Request<{ categoryId: string }, {}, {}, { pageSize: string; page: string }>,
+			res: Response,
+			next: NextFunction,
+		) => {
+			try {
+				const categoryId = Number(req.params.categoryId);
+				let page = parseInt(req.query.page as string) || defaultPage;
+				let pageSize = parseInt(req.query.pageSize as string) || defaultPageSize;
+
+				logger.info(
+					LOG_MODULE,
+					`Request Submitted Data on categoryId '${categoryId}'`,
+					`pagination params: page '${page}' pageSize '${pageSize}'`,
+				);
+
+				if (page < 0) {
+					throw new BadRequest('Invalid `page` parameter. Expecting a numeric value greater than 0');
+				}
+
+				if (pageSize < 0) {
+					throw new BadRequest('Invalid `pageSize` parameter. Expecting a numeric value greater than 0');
+				}
+
+				if (_.isNaN(categoryId)) {
+					throw new BadRequest('Invalid categoryId number format');
+				}
+
+				const submittedDataResult = await service.getSubmittedDataByCategory(categoryId, { page, pageSize });
+
+				if (_.isEmpty(submittedDataResult.data)) throw new NotFound('No Submitted Data found');
+
+				return res.status(200).send({
```

Comment:
rather than asserting the type, you could instantiate the response before sending it. e.g.
```js
const response: SubmittedDataPaginatedResponse = {...};

return res.status(200).send(response);
```

Reply:
Declaring over asserting 👍